train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java

```diff
@@ -23,6 +23,13 @@ public void followSpeed() {
 		enforceSpeedLimit();
 	}
 
+	@Override
```

Comment:
The new run() function implementation might cause trouble because while is an endless loop